tools/AutomationTestFramework/Management/PackageGenerator.ps1

```diff
@@ -0,0 +1,141 @@
+    $null = $files.Add((Copy-Item "$rmCommonDir\$commonFileName" "$archiveDir\Very$commonFileName" -PassThru -ErrorAction Stop))
+
+    # Create the module file
+    $moduleFile = New-Item "$archiveDir\$moduleName.psm1" -type file
+    $files | Where-Object { $_.Name -like '*.ps1' } `
+        | ForEach-Object {'. "$PSScriptRoot\' + $_.Name + '"'} `
+        | Add-Content $moduleFile.FullName -ErrorAction Stop
+    $null = $files.Add($moduleFile)
+
+    $zipPath = "$archiveDir\$moduleName.zip"
+    if(Test-Path $zipPath) {
+        Remove-Item $zipPath -Force -ErrorAction Stop
+    }
+    Write-Verbose "Creating $zipPath..."
+    $files | Compress-Archive -CompressionLevel Fastest -DestinationPath $zipPath -ErrorAction Stop
+    Write-Verbose "$zipPath created."
+
+    Get-ChildItem -Path $archiveDir -Filter "*.ps*" | Remove-Item -Force -ErrorAction Stop
+}
+
+# Remove version and rename extension .nupkg -> .zip
+# Example: AzureRM.Compute.3.3.2.nupkg -> AzureRM.Compute.zip
+function Convert-NupkgToZip (
+    [string] $path,
+    [string[]] $moduleList,
+    [string] $outputPath) {
+   
+    if (-not (Test-Path $outputPath)) {
+        $null = New-Item -ItemType directory -Path $outputPath -ErrorAction Stop
+    }
+
+    $modulePathFilters = $moduleList | ForEach-Object { 
+        $nupkgPath = Join-Path $path "${_}*.nupkg" 
+        @{Name = $_; Path = $nupkgPath; Exists = Test-Path $nupkgPath}
+    }
+    # https://stackoverflow.com/a/26559478/294804
+    $notFound = ($modulePathFilters | ForEach-Object { $_.Exists }) -contains $false
+    if ($notFound) {
+        $missingModules = $modulePathFilters | Where-Object { -not $_.Exists } | ForEach-Object { $_.Name }
+        throw "Cannot find modules ($missingModules) in the directory '$path'"
+    }
+
+    $packages = $modulePathFilters | ForEach-Object { @{Name = $_.Name; File = (Get-ChildItem $_.Path | Select-Object -Last 1)} }
+    foreach ($package in $packages) {
+        $zipPath = Join-Path $outputPath "$($package.Name).zip"
+        if(Test-Path $zipPath) {
+            Remove-Item $zipPath -Force -ErrorAction Stop
+        }
+        Copy-Item $package.File.FullName $zipPath
+    }
+}
+
+function Get-LatestBuildPath([string] $searchPath) {
+    $folderSuffix = '_PowerShell'
+    # https://stackoverflow.com/a/25165057/294804
+    Get-ChildItem $searchPath -Directory -Filter "*$folderSuffix" `
+        | ForEach-Object -Begin { 
+            $date = [DateTime]::MinValue.ToString('yyyy_MM_dd')
+            $path = $null
+        } -Process { 
+            $folderDateText = $_.Name -ireplace "$folderSuffix`$", ''
+            if($folderDateText -gt $date) { 
+                $date = $folderDateText
+                $path = $_
+            }
+        } -End { $path }
+}
+
+function Create-SignedModules([hashtable] $signedModules, [string] $archiveDir) {
+    $latestBuildPath = Get-LatestBuildPath -searchPath '\\aaptfile01\ADXSDK\PowerShell'
```

Comment:
Can we make this drop location a script parameter?

Reply:
The changes to this are in the next PR/branch, Test On Build. That primarily is for allowing this process to work off of the location of drops. This \\aaptfile01\ process becomes the default if no signed drop location is passed in (on that branch).